### PR TITLE
FE: Monitoring Policies - more unit tests for Monitoring Policies components

### PR DESCRIPTION
**Pull request by faizanjanjua**
## Description
Add more unit tests for Monitoring Policies components. For example, clicking on an item or button updates the monitoringPoliciesStore correctly or navigates correctly. Some examples:

clicking on an item in the table opens the Details panel

clicking Save saves the policy - or displays an error message if Monitoring Policy or Rule name is missing

clicking Edit, Copy, Delete in the Details panel results in the appropriate action and changes in store

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2566

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

File: ui/tests/MonitoringPolicies/MonitoringPolicyDetailPanel.test.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    await router.isReady()
+    expect(router.currentRoute.value.params.id).toBe(mockMonitoringPolicy.id)
+    expect(store.policyEditMode).toBe(CreateEditMode.Edit)
+  })
+  
+  test('The Copy button should call onCopy method', async () => {
+    const copyButton = wrapper.get('[data-test="onCopy"]')
+    await copyButton.trigger('click')
+    let store = useMonitoringPoliciesStore()
+    expect(store.copyPolicy).toBeCalledWith(mockMonitoringPolicy)
```
Comment (Shahbaz-dataq) [inline ui/tests/MonitoringPolicies/MonitoringPolicyDetailPanel.test.ts:65]: You can also check if `selectedPolicy.id` toBe `null` and `selectedPolicy.name` toBe `Copy of ${mockMonitoringPolicy.name}`

File: ui/tests/MonitoringPolicies/MonitoringPolicyDetailPanel.test.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  })
+
+  test('The Close button should call onClose method', async () => {
+    const closeButton = wrapper.findComponent('[data-test="onClose"]')
+    await closeButton.trigger('click')
+    let store = useMonitoringPoliciesStore()
+    expect(store.clearSelectedPolicy).toBeCalled()
```
Comment (Shahbaz-dataq) [inline ui/tests/MonitoringPolicies/MonitoringPolicyDetailPanel.test.ts:72]: Make it `toBeCalledOnce`.
Also, add `expect(store.clearSelectedRule).toBeCalledOnce()`

File: ui/tests/MonitoringPolicies/MonitoringPolicyDetailPanel.test.ts [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  })
+
+  test('The alert rules should be displayed correctly', () => {
+    const rules = wrapper.findAll('.alert-rule')
+      // @ts-ignore 
+    expect(rules.length).toBe(mockMonitoringPolicy.rules.length)
+      // @ts-ignore 
+    rules.forEach((ruleWrapper, index) => {
+        // @ts-ignore 
+    expect(ruleWrapper.text()).toContain(mockMonitoringPolicy.rules[index].name)
```
Comment (Shahbaz-dataq) [inline ui/tests/MonitoringPolicies/MonitoringPolicyDetailPanel.test.ts:82]: Indent 1 space

File: ui/tests/MonitoringPolicies/MonitoringPolicyDetailPanel.test.ts [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+  test('The Save button should be disabled when the form is invalid', async () => {
+    const saveButton = wrapper.getComponent({ name: 'ButtonWithSpinner' })
+    await saveButton.trigger('click.prevent')
+    expect(wrapper.vm.isSavePolicyDisabled).toBe(true)
+  })
+
+  test('The Save button should be enabled when the form is valid', async () => {
+    const mockMonitoringPolicyNew: Policy = cloneDeep(mockMonitoringPolicy)
+    mockMonitoringPolicyNew.rules   =  [
+    {
```
Comment (Shahbaz-dataq) [inline ui/tests/MonitoringPolicies/MonitoringPolicyDetailPanel.test.ts:101]: Indent 1 space for whole object

File: ui/tests/MonitoringPolicies/MonitoringPolicyDetailPanel.test.ts [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+  test('The Save button should be disabled when the form is invalid', async () => {
+    const saveButton = wrapper.getComponent({ name: 'ButtonWithSpinner' })
+    await saveButton.trigger('click.prevent')
+    expect(wrapper.vm.isSavePolicyDisabled).toBe(true)
+  })
+
+  test('The Save button should be enabled when the form is valid', async () => {
+    const mockMonitoringPolicyNew: Policy = cloneDeep(mockMonitoringPolicy)
+    mockMonitoringPolicyNew.rules   =  [
```
Comment (Shahbaz-dataq) [inline ui/tests/MonitoringPolicies/MonitoringPolicyDetailPanel.test.ts:100]: Remove extra spaces

File: ui/tests/MonitoringPolicies/MonitoringPolicyDetailPanel.test.ts [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+    directives: {
+      focus: featherInputFocusDirective
+    },
+    router : router
+  }
+})
+
+describe('Monitoring Policies', () => {
+
+beforeEach(()=>{
```
Comment (Shahbaz-dataq) [inline ui/tests/MonitoringPolicies/MonitoringPolicyDetailPanel.test.ts:38]: This whole function need indenting:
for `beforeEach` function indent 2 spaces and for internal function make indent of 4 spaces also add spaces between `()=>{`

File: ui/src/components/MonitoringPolicies/MonitoringPoliciesDetailPanel.vue [hunk 7]
```diff
@@ -158,10 +160,12 @@ const removePolicy = () => {
 }
 
 const updateMonitoringPolicyStatus = (newStatus: boolean) => {
+  console.log('newStatus', newStatus)
   isPolicyEnabled.value = newStatus
 }
 
 const savePolicy = async () => {
+  console.log('savePolicy')
```
[the remark this answers is not in the document]
Comment (faizanjanjua) [reply ui/src/components/MonitoringPolicies/MonitoringPoliciesDetailPanel.vue:168]: resolved thanks

**Review by Shahbaz-dataq — COMMENTED** [review]
Approve

**Review by synqotik — APPROVED** [review]
LGTM

**Pull request merged**